fastlane/Fastfile

```diff
@@ -38,80 +68,182 @@ platform :ios do
+    )
+
+    # Get rollbar server access token from configuration file.
+    rollbar_server_access_token = read_xcconfig_property(
+      xcconfig_path: project_xcconfig_path % [ENV['PROJECT_NAME'], build_configurations[:testflight]],
+      xcconfig_key: 'ROLLBAR_SERVER_ACCESS_TOKEN'
+    )
+
+    # Upload dsym to rollbar.
+    upload_dsym(
+      access_token: rollbar_server_access_token,
+      version: next_build_number,
+      bundle_identifier: bundle_identifier,
+      dsym_zip_path: dsym_zip_path % ENV['PROJECT_NAME']
+    )
+
+    # Puts back the backed up build number in Info.plist.
+    set_info_plist_value(
+      path: project_plist_path % ENV['PROJECT_NAME'],
+      key: 'CFBundleVersion',
+      value: info_plist_backup_build_number
     )
 
     publish_testflight
+  end
+
+  lane :test do
+    # Run scan with default project and scheme
+    scan(
+      configuration: build_configurations[:test],
+      clean: false
+    )
 
   end
 
-  private_lane :build_app do |options|
+  lane :create_development_app do
+    # Remember after this point to choose this profile in xCode Signing (Development)
+    create_app(
+      app_name: app_names[:development] % ENV['PROJECT_NAME'],
+      build_configuration: build_configurations[:development],
+      team_name: ENV['TEAM_NAME'],
+      skip_itc: true,
+      match_type: match_types[:development],
+    )
+
+    # Remember after this point to choose this profile in xCode Signing (Alpha)
+    create_app(
+      app_name: app_names[:qa] % ENV['PROJECT_NAME'],
+      build_configuration: build_configurations[:qa],
+      team_name: ENV['TEAM_NAME'],
+      skip_itc: true,
+      match_type: match_types[:qa],
+    )
+
+  end
+
+  lane :create_testflight_app do
+    # Remember after this point to choose this profile in xCode Signing (Beta)
+    create_app(
+      app_name: app_names[:testflight] % ENV['PROJECT_NAME'],
+      build_configuration: build_configurations[:testflight],
+      team_name: ENV['TEAM_NAME'],
+      skip_itc: true,
+      match_type: match_types[:testflight],
+    )
+
+  end
+
+  lane :add_device do
+    device_name = prompt(text: 'Enter the device name: ')
+    device_udid = prompt(text: 'Enter the device UDID: ')
+    device_hash = {}
+    device_hash[device_name] = device_udid
+
+    # Register new device.
+    register_devices(
+      devices: device_hash
+    )
 
-    # Update the app identifier
-    update_app_identifier(
-      xcodeproj: "#{options[:project_name]}.xcodeproj",
-      plist_path: "#{options[:project_name]}/Info.plist",
-      app_identifier: options[:app_identifier]
+    # Refresh provisioning profiles adding the new device.
+    match(
+      force_for_new_devices: true
     )
 
-    # Download certificates
-    cert
+  end
+
+  private_lane :build_app do |options|
     # Download provisioning profiles
-    sigh
+    match(
+      app_identifier: options[:app_identifier],
+      type: options[:match_type],
+      readonly: true
```

Comment:
Hmm but why would it be a mess? Now that we have corrected the team branch we've been maintaining only one Wolox S.A. cert, and we haven't generated duplicates since :tada:. 
Worst case scenario, you needed to create again the provisioning because it expired or it became invalid for everyone for some reason, right? My point is, either way we'll end up running the lane where this is set to `true` to generate a new PP.